Clang static analyzer reports a warning on qi_real.hpp

Clang static analyzer gives me the following warning.

```c++
In file included from include/boost/spirit/include/qi_real.hpp:16:
In file included from include/boost/spirit/home/qi/numeric/real.hpp:21:
include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:318:31: warning: Function call argument is an uninitialized value
            traits::assign_to(traits::negate(neg, n), attr);
                              ^~~~~~~~~~~~~~~~~~~~~~
```

I cannot tell if it is a false-positive or if this can be silenced.

Here is a full report:
```
1 warning generated.
/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:318:31: warning: 2nd function call argument is an uninitialized value [clang-analyzer-core.CallAndMessage]
            traits::assign_to(traits::negate(neg, n), attr);
                              ^
cpp/test_spirit/main.cpp:20:10: note: Calling 'my_parse'
  return my_parse(str, qi::double_, val);
         ^
cpp/test_spirit/main.cpp:10:12: note: Calling 'parse'
  bool r = qi::parse(it, str.end(), parser, val);
           ^
/include/boost/spirit/home/qi/parse.hpp:86:16: note: Calling 'any_real_parser::parse'
        return compile<qi::domain>(expr).parse(first, last, context, unused, attr);
               ^
/include/boost/spirit/home/qi/numeric/real.hpp:168:20: note: Calling 'real_impl::parse'
            return extract::parse(first, last, attr_, RealPolicies());
                   ^
/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:188:13: note: Taking false branch
            if (first == last)
            ^
/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:197:13: note: 'n' declared without an initial value
            T n;
            ^
/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:204:13: note: Taking false branch
            if (!got_a_number)
            ^
/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:229:13: note: Taking true branch
            if (p.parse_dot(first, last))
            ^
/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:234:17: note: Taking true branch
                if (p.parse_frac_n(first, last, acc_n, frac_digits))
                ^
/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:270:13: note: Taking true branch
            if (e_hit)
            ^
/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:275:17: note: Taking false branch
                if (p.parse_exp_n(first, last, exp))
                ^
/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:288:21: note: Calling 'scale'
                    traits::scale(-frac_digits, n, acc_n);
                    ^
/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:66:13: note: Assuming 'exp' is >= 0
        if (exp >= 0)
            ^
/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:66:9: note: Taking true branch
        if (exp >= 0)
        ^
/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:72:17: note: Calling 'integral_constant::operator bool'
            if (is_floating_point<T>() && exp > max_exp)
                ^
/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:72:17: note: Returning from 'integral_constant::operator bool'
/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:72:17: note: Left side of '&&' is true
/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:72:43: note: Assuming 'exp' is > 'max_exp'
            if (is_floating_point<T>() && exp > max_exp)
                                          ^
/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:72:13: note: Taking true branch
            if (is_floating_point<T>() && exp > max_exp)
            ^
/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:288:21: note: Returning from 'scale'
                    traits::scale(-frac_digits, n, acc_n);
                    ^
/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp:318:31: note: 2nd function call argument is an uninitialized value
            traits::assign_to(traits::negate(neg, n), attr);
                              ^
```

My test program:
```c++
#include <boost/spirit/include/qi_real.hpp>
#include <boost/spirit/home/qi/parser.hpp>

namespace qi = boost::spirit::qi;

template <typename P, typename T>
bool my_parse(std::string const& str, P&& parser, T& val)
{
  auto it = str.begin();
  bool r = qi::parse(it, str.end(), parser, val);

  if (it != str.end()) return false;
  return r;
}

int main()
{
  double val;
  std::string str = "1." + std::string(309, '3') + "e";
  return my_parse(str, qi::double_, val);
}

```

--- a/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp
+++ b/include/boost/spirit/home/qi/numeric/detail/real_impl.hpp
@@ -234,6 +234,7 @@ namespace boost { namespace spirit { namespace qi  { namespace detail
                 // to zero (0) only if we already got a number.
                 if (p.parse_frac_n(first, last, acc_n, frac_digits))
                 {
+                    BOOST_ASSERT(frac_digits >= 0);
                 }
                 else if (!got_a_number || !p.allow_trailing_dot)
                 {
@@ -286,13 +287,15 @@ namespace boost { namespace spirit { namespace qi  { namespace detail
                     // by resetting 'first' prior to the exponent prefix (e|E)
                     first = e_pos;
                     // Scale the number by -frac_digits.
-                    traits::scale(-frac_digits, n, acc_n);
+                    bool r = traits::scale(-frac_digits, n, acc_n);
+                    BOOST_VERIFY(r);
                 }
             }
             else if (frac_digits)
             {
                 // No exponent found. Scale the number by -frac_digits.
-                traits::scale(-frac_digits, n, acc_n);
+                bool r = traits::scale(-frac_digits, n, acc_n);
+                BOOST_VERIFY(r);
             }
             else if (traits::is_equal_to_one(acc_n))
             {

--- a/include/boost/spirit/home/x3/support/numeric_utils/extract_real.hpp
+++ b/include/boost/spirit/home/x3/support/numeric_utils/extract_real.hpp
@@ -180,6 +180,7 @@ namespace boost { namespace spirit { namespace x3
                     if (!is_same<T, unused_type>::value)
                         frac_digits =
                             static_cast<int>(std::distance(savef, first));
+                    BOOST_ASSERT(frac_digits >= 0);
                 }
                 else if (!got_a_number || !p.allow_trailing_dot)
                 {

